### PR TITLE
update to vue 3, vite, sdk 13, refactor components, store

### LINKED ISSUE #78: ACI - show the format returned by compiler
## Current output
```json
{
  "encodedAci": {
    "contract": {
      "functions": [
        {
          "arguments": [
            {
              "name": "x",
              "type": "int"
            }
          ],
          "name": "example",
          "payable": false,
          "returns": "int",
          "stateful": false
        }
      ],
      "kind": "contract_main",
      "name": "Example",
      "payable": false,
      "type_defs": []
    }
  },
  "externalEncodedAci": []
}
```

## Expected output
```json
{
   "encoded_aci":{
      "contract":{
         "functions":[
            {
               "arguments":[
                  {
                     "name":"x",
                     "type":"int"
                  }
               ],
               "name":"example",
               "payable":false,
               "returns":"int",
               "stateful":false
            }
         ],
         "kind":"contract_main",
         "name":"Example",
         "payable":false,
         "type_defs":[
            
         ]
      }
   },
   "external_encoded_aci":[
      
   ],
   "interface":"main contract Example =\n  entrypoint example : (int) => int\n"
}
```

as far as I remember we agreed that we want to use the ACI version returned by the http compiler instead of using the version translated by the SDK. cc @hanssv @kenodressel @thepiwo @kalovelo @nikita-fuchs @radrow 

also I am not sure about the reason to drop the `interface`. this was mainly requested by @thepiwo. I didn't remove it now in the example here. just wanted to make sure we agree on dropping it. in the SDK we could just ignore the `interface` if not needed (cc @davidyuk)

### LINKED ISSUE #87: Update to Vue 3, Vite, SDK 13

### LINKED ISSUE #91: adjustable editor input size
currently the input field of the editor is fixed. would be good being able to edit (increase) its size

### PR DESCRIPTION
**Pull request by thepiwo**
closes #87
closes #86 
closes #78
closes #91

### CODE REVIEW

File: src/components/Account.vue [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+      @click="initSdk(false)"
+    >
+      Connect Wallet Extension
+    </button>
+    <button
+      class="mt-2 mr-4 rounded-full bg-black hover:bg-purple-500 text-white p-2 px-4"
+      @click="modifySettings = !modifySettings"
+    >
+      <span v-if="isStatic">Modify Local Account</span>
+      <span v-if="!isStatic">Use Local Account</span>
```
Comment (lukeromanowicz) [inline src/components/Account.vue:15]: You can use v-else here :)

File: src/components/Account.vue [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        >
+          Use Local Account
+        </button>
+      </div>
+    </div>
+  </div>
+
+  <h1 class="py-2">
+    Test contracts
+    <span v-if="!isConnected && !clientError" class="text-sm text-red-500">
```
Comment (lukeromanowicz) [inline src/components/Account.vue:62]: looks like all thre error spans are styled the same way. In such case I would combine them into one while the message could be a computed property that gets the value set by the logic you implemented in the ifs

**Review by lukeromanowicz — APPROVED** [review]
Really good job!! I didn't find any major issues while the volume of work and level of maintainability improvement are both enormous!

File: src/components/Account.vue (hunk 1 above)
Comment (thepiwo) [reply src/components/Account.vue:15]: yeah its copied from legacy, I'll adjust

File: src/components/Account.vue (hunk 2 above)
Comment (thepiwo) [reply src/components/Account.vue:62]: yeah its copied from legacy, I'll adjust

**Comment by thepiwo** [comment]
@davidyuk @kenodressel ready for final review if you wish

File: src/stores/contractStore.ts [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+      .compileBySourceCode(compileData.value.contractCode)
+      .then((result) => {
+        compileResult.value.setFinal("Compiled from source", {
+          aci: JSON.stringify(result?.aci, null, 2),
+          byteCode: result.bytecode,
+        });
+
+        persistContract(
+          compileData.value.contractCode,
+          compileResult.value.data?.aci || "",
```
Comment (kenodressel) [inline src/stores/contractStore.ts:80]: why is the ACI optional? is that not a reason to error?

File: src/stores/contractStore.ts [hunk 4]
```diff
@@ -0,0 +1,275 @@
...
+    contractAddress: string,
+    aci: string
+  ) {
+    resetDeployAndCallData();
+    deployResult.value.setInfo("Instantiating Contract at address ...");
+
+    await sdkStore.aeSdk
+      ?.initializeContract({
+        ...{ aci: JSON.parse(aci) },
+        address: `ct_${contractAddress.replace("ct_", "")}`,
```
Comment (kenodressel) [inline src/stores/contractStore.ts:103]: what does conversion do? removing something just to add it back in the same place? or are you unsure about the format you receive the address in?

File: src/stores/contractStore.ts [hunk 5]
```diff
@@ -0,0 +1,275 @@
...
+    const options = Object.fromEntries(
+      Object.entries(deployData.value.options).filter(([_, v]) => v != null)
+    );
+
+    contractInstance
+      ?.$deploy(args, options)
+      .then((deployed) => {
+        deployResult.value.setFinal(
+          `Deployed, and mined at this address: ${deployed?.result?.contractId}`,
+          deployed?.result?.contractId
```
Comment (kenodressel) [inline src/stores/contractStore.ts:145]: should not be optional, right?

File: src/stores/contractStore.ts [hunk 6]
```diff
@@ -0,0 +1,275 @@
...
+
+    contractInstance
+      ?.$call(callStaticData.value.func, args, options)
+      .then((result) => {
+        callStaticResult.value.setFinal(
+          `Dry-Run Gas Estimate: ${
+            result?.result?.gasUsed
+            // @ts-ignore
+          }, Fee Estimate: ${toAe(result?.tx?.fee)} ae (${
+            // @ts-ignore
```
Comment (kenodressel) [inline src/stores/contractStore.ts:171]: why ts-ignore here?

File: src/stores/sdkStore.ts [hunk 7]
```diff
@@ -0,0 +1,200 @@
...
+        const handleWallets: ({
+          wallets,
+          newWallet,
+        }: {
+          wallets: Wallets;
+          newWallet?: Wallet | undefined;
+        }) => Promise<void> = async ({ wallets, newWallet }) => {
+          newWallet = newWallet || Object.values(wallets)[0];
+          stopScan();
+          if (!aeSdk) return;
```
Comment (kenodressel) [inline src/stores/sdkStore.ts:79]: reject?

File: src/stores/sdkStore.ts [hunk 8]
```diff
@@ -0,0 +1,200 @@
...
+
+          try {
+            (aeSdk as AeSdkAepp).disconnectWallet();
+          } catch (_) {}
```
Comment (kenodressel) [inline src/stores/sdkStore.ts:83]: reject?

File: src/stores/sdkStore.ts [hunk 9]
```diff
@@ -0,0 +1,200 @@
...
+          );
+
+          aeSdk.selectNode(networkId);
+
+          await (aeSdk as AeSdkAepp).subscribeAddress(
+            SUBSCRIPTION_TYPES.subscribe,
+            "current"
+          );
+
+          resolve("");
```
Comment (kenodressel) [inline src/stores/sdkStore.ts:96]: `resolve()`?

File: src/stores/sdkStore.ts [hunk 10]
```diff
@@ -0,0 +1,200 @@
...
+
+      await updateConnectionInfo();
+    } catch (e) {
+      status.value = Status.CONNECTION_ERROR;
+    }
+  }
+
+  async function updateConnectionInfo() {
+    status.value = Status.FETCHING_INFO;
+    networkId.value = await aeSdk?.api?.getNetworkId();
```
Comment (kenodressel) [inline src/stores/sdkStore.ts:158]: should we reject here when aeSdk is not defined?

**Review by kenodressel — COMMENTED** [review]
sorry for the amount of comments but it seems like we could use this as a reference for other projects so I was a bit detailed :sweat_smile:

File: src/stores/sdkStore.ts (hunk 7 above)
Comment (thepiwo) [reply src/stores/sdkStore.ts:79]: actually not needed any more, the check is already on the outside scope

File: src/stores/sdkStore.ts (hunk 8 above)
Comment (thepiwo) [reply src/stores/sdkStore.ts:83]: no, we specifically ignore, to ensure we are disconnected before trying to connect, but disconnect will error if we are not connected, adding a comment

File: src/stores/sdkStore.ts (hunk 9 above)
Comment (thepiwo) [reply src/stores/sdkStore.ts:96]: yes, needed to specify `Promise<void>` to make eslint happy

File: src/stores/sdkStore.ts (hunk 10 above)
Comment (thepiwo) [reply src/stores/sdkStore.ts:158]: didn't see a good way to do it and currently the case won't happen as far as I know

File: src/stores/contractStore.ts (hunk 5 above)
Comment (thepiwo) [reply src/stores/contractStore.ts:145]: seems to be fixed in relased sdk

File: src/stores/contractStore.ts (hunk 4 above)
Comment (thepiwo) [reply src/stores/contractStore.ts:103]: was an issue in the sdk beta, seems to be fixed now

File: src/components/Account.vue [hunk 11]
```diff
@@ -0,0 +1,100 @@
+<template>
+  <div class="flex">
+    <button
+      v-if="isStatic"
+      class="mt-2 mr-2 rounded-full bg-black hover:bg-purple-500 text-white p-2 px-4"
```
[the remark this answers is not in the document]
Comment (thepiwo) [reply src/components/Account.vue:5]: didn't want to extend the refactoring even further, in the future if we want to continue active work on this project, yes

File: src/components/Account.vue [hunk 12]
```diff
@@ -0,0 +1,100 @@
...
+    >
+      <span v-if="isStatic">Modify Local Account</span>
+      <span v-else>Use Local Account</span>
+    </button>
+    <h6 v-if="!modifySettings && address" class="mt-4 text-sm text-purple">
+      <span class="font-mono text-black">Account: </span> {{ address }}
+    </h6>
+  </div>
+
+  <div v-if="!secretKey || !address || !nodeUrl || modifySettings">
```
[the remark this answers is not in the document]
Comment (thepiwo) [reply src/components/Account.vue:22]: do you have a specific proposal?

**Review by marc0olo — APPROVED** [review]
additional request from my side:
- please allow the user to increase the window of the editor (currently only possible for the console output)

@thepiwo let me know if I should create an issue

**Comment by marc0olo** [comment]
@kenodressel can you take another look? would be nice if we can release this asap. wanna share a tooling update with hackathon participants

**Comment by marc0olo** [comment]
@thepiwo created a new issue for the topic mentioned above: https://github.com/aeternity/aepp-contracts/issues/91

File: src/components/Account.vue (hunk 12 above)
[the remark this answers is not in the document]
Comment (kenodressel) [reply src/components/Account.vue:22]: secretKey && address && nodeUrl && !modifySettings is IMHO much easier to read

**Comment by thepiwo** [comment]
@marc0olo good catch, fixed it

File: src/components/Account.vue (hunk 12 above)
[the remark this answers is not in the document]
Comment (thepiwo) [reply src/components/Account.vue:22]: as discussed its fine as is

**Review by kenodressel — APPROVED** [review]

**Pull request merged**